dipy/reconst/ivim.py

```diff
@@ -364,20 +368,28 @@ def estimate_linear_fit(self, data, split_b, less_than=True):
 
         D : float
             The estimated value of D.
+
+        References
+        ----------
+        Lawson C., Hanson R.J., (1987) Solving Least Squares Problems, SIAM
+
         """
         if less_than:
             bvals_split = self.gtab.bvals[self.gtab.bvals <= split_b]
-            D, neg_log_S0 = np.polyfit(bvals_split,
-                                       -np.log(data[self.gtab.bvals <=
-                                                    split_b]), 1)
+            b = -np.log(data[self.gtab.bvals <= split_b])
+
+            D = nnls(np.matrix(bvals_split).T, b)[0]
+            _, neg_log_S0 = np.polyfit(bvals_split, b, 1)
+
         else:
             bvals_split = self.gtab.bvals[self.gtab.bvals >= split_b]
-            D, neg_log_S0 = np.polyfit(bvals_split,
-                                       -np.log(data[self.gtab.bvals >=
-                                                    split_b]), 1)
+            _b = -np.log(data[self.gtab.bvals >= split_b])
+
+            D = nnls(np.matrix(bvals_split).T, _b)[0]
```

Comment:
Do you need the `[0]` both here and below on line 392?

Reply:
Yes, so... D returned by `nnls` is a single element `array`. with `D[0]` I am picking out that element to ease the `IvimFit`. Does that make sense?